pkg/operator/controller/status.go

```diff
@@ -95,123 +94,147 @@ func (r *reconciler) syncOperatorStatus() error {
 
 // getOperatorState gets and returns the resources necessary to compute the
 // operator's current state.
-func (r *reconciler) getOperatorState() (*corev1.Namespace, []operatorv1.DNS, []appsv1.DaemonSet, error) {
+func (r *reconciler) getOperatorState() (*corev1.Namespace, []operatorv1.DNS, error) {
 	ns := manifests.DNSNamespace()
 	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: ns.Name}, ns); err != nil {
 		if errors.IsNotFound(err) {
-			return nil, nil, nil, nil
+			return nil, nil, nil
 		}
-		return nil, nil, nil, fmt.Errorf("failed to get namespace %s: %v", ns.Name, err)
+		return nil, nil, fmt.Errorf("failed to get namespace %s: %v", ns.Name, err)
 	}
 
 	dnsList := &operatorv1.DNSList{}
 	if err := r.client.List(context.TODO(), dnsList); err != nil {
-		return nil, nil, nil, fmt.Errorf("failed to list dnses: %v", err)
+		return nil, nil, fmt.Errorf("failed to list dnses: %v", err)
 	}
 
-	daemonsetList := &appsv1.DaemonSetList{}
-	if err := r.client.List(context.TODO(), daemonsetList, client.InNamespace(ns.Name)); err != nil {
-		return nil, nil, nil, fmt.Errorf("failed to list daemonsets: %v", err)
+	return ns, dnsList.Items, nil
+}
+
+// computeOperatorStatusConditions computes the operator's current state.
+func computeOperatorStatusConditions(oldConditions []configv1.ClusterOperatorStatusCondition, ns *corev1.Namespace,
+	dnsesAvailable, dnses int) []configv1.ClusterOperatorStatusCondition {
+	var oldDegradedCondition, oldProgressingCondition, oldAvailableCondition *configv1.ClusterOperatorStatusCondition
+	for i := range oldConditions {
+		switch oldConditions[i].Type {
+		case configv1.OperatorDegraded:
+			oldDegradedCondition = &oldConditions[i]
+		case configv1.OperatorProgressing:
+			oldProgressingCondition = &oldConditions[i]
+		case configv1.OperatorAvailable:
+			oldAvailableCondition = &oldConditions[i]
+		}
 	}
-	return ns, dnsList.Items, daemonsetList.Items, nil
+
+	conditions := []configv1.ClusterOperatorStatusCondition{}
+	conditions = append(conditions, computeOperatorDegradedCondition(oldDegradedCondition, dnsesAvailable, dnses, ns))
+	conditions = append(conditions, computeOperatorProgressingCondition(oldProgressingCondition, dnsesAvailable, dnses))
+	conditions = append(conditions, computeOperatorAvailableCondition(oldAvailableCondition, dnsesAvailable))
```

Comment:
In my earlier comment about using a slice literal, I meant to suggest the following:
```go
	conditions := []configv1.ClusterOperatorStatusCondition{
		computeOperatorDegradedCondition(oldDegradedCondition, dnsesAvailable, dnses, ns),
		computeOperatorProgressingCondition(oldProgressingCondition, dnsesAvailable, dnses),
		computeOperatorAvailableCondition(oldAvailableCondition, dnsesAvailable),
	}
￼```